Gateway integration tests

werdes72 opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines, templates, the recommended Git workflow, and any related documentation.
2. Read and submit the required Contributor Licence Agreements (https://github.com/kyma-project/community/blob/main/docs/contributing/02-contributing.md#agreements-and-licenses).
3. Test your changes and attach their results to the pull request.
4. Update the relevant documentation.
-->

**Description**

Changes proposed in this pull request:

- Add integration tests with gateway CR

jaroslaw-pieszka committed:
sed corrected, additional target to get latest release version (#553)

nataliasitko committed:
API Gateway documentation (#521)

* add initial api gateway docu

* update documentation

* review docs

* fix links

* revert unnecessary changes

* fix links

* add apirule readme

* language fixes

* review docs

* adjust diagrams

* Remove ClusterRoles

* apply Bartek's suggestions

* remove outdated tutorial

* remove serverless dep

* remove serverless deps

* remove serverless deps

* remove titles

* remove block list tg

* Update docs/user/technical-reference/05-00-api-gateway-operator-parameters.md

Co-authored-by: Bartosz Chwila <103247439+barchw@users.noreply.github.com>

* delete blocklist docu

* fix links

* remove-cipher-suites

* remove articles

* remove minProtocolVersions and cipherSuites from examples

* Apply suggestions from code review

Co-authored-by: Grzegorz Karaluch <grzegorz.karaluch@sap.com>

* Apply suggestions from code review

Co-authored-by: Grzegorz Karaluch <grzegorz.karaluch@sap.com>

* restructure 404 tutorial

* language fixes

* language fixes

* update-links

---------

Co-authored-by: Bartosz Chwila <103247439+barchw@users.noreply.github.com>
Co-authored-by: Grzegorz Karaluch <grzegorz.karaluch@sap.com>

werdes72 committed:
Integration tests with gateway CR

werdes72 committed:
Integration tests with gateway CR

werdes72 committed:
Restore config files

werdes72 committed:
Restore config files

werdes72 committed:
Don't use namespace

werdes72 committed:
Don't use namespace

werdes72 committed:
Add gateway removal after test suite run

werdes72 committed:
Add gateway removal after test suite run

triffer committed:
Apply documentation suggestions from code review

Co-authored-by: Bartosz Chwila <103247439+barchw@users.noreply.github.com>
Co-authored-by: Natalia Sitko <80401180+nataliasitko@users.noreply.github.com>

Commit:
Update diagrams with diagrams proposed in review

Commit:
Merge branch 'mod-dev' into default-gateway

Commit:
Code review remarks

Commit:
Merge temporary doc into real doc

werdes72 committed:
Wait for gateway and check its state

werdes72 committed:
Wait for gateway and check its state

kolodziejczak committed:
Adjust custom domain test's deploy to gardener (#565)

* Adjust custom domain test's deploy to gardener

* Improve comment

videlov commented on tests/integration/manifests/api-gateway.yaml:
```diff
@@ -0,0 +1,6 @@
+apiVersion: operator.kyma-project.io/v1alpha1
```
could you move this manifests folder under `testsuites/gateway` like other testsuites ?

videlov commented on tests/integration/main_test.go:
```diff
@@ -73,13 +77,35 @@ func TestOryJwt(t *testing.T) {
 		// We are not using ScenarioInitializer, as this function only needs to set up global resources
 		TestSuiteInitializer: func(ctx *godog.TestSuiteContext) {
+			ctx.BeforeSuite(func() {
+				if err := createApiGatewayCR(config); err != nil {
```
shouldn't we do this part of the scenarios, like for istio?, e.g.:
https://github.com/kyma-project/istio/blob/66911ce5bb2d2dfde90fb1401f377868769ca9b9/tests/integration/features/istio/evaluation/istio_installation_demo.feature#L2-L6

tearing down with the hooks approach as well:
https://github.com/kyma-project/istio/blob/66911ce5bb2d2dfde90fb1401f377868769ca9b9/tests/integration/hooks.go#L18-L69

dependabot[bot] committed:
gomod(deps): bump github.com/onsi/gomega from 1.27.10 to 1.28.0 (#571)

Bumps [github.com/onsi/gomega](https://github.com/onsi/gomega) from 1.27.10 to 1.28.0.
- [Release notes](https://github.com/onsi/gomega/releases)
- [Changelog](https://github.com/onsi/gomega/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/gomega/compare/v1.27.10...v1.28.0)

---
updated-dependencies:
- dependency-name: github.com/onsi/gomega
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kolodziejczak committed:
Integration tests pipelines (#572)

* Save point

* Save point 2

* change deploy script controller to manager

* save point again

* Cleanup

* Remove unused var in tests

* change workflow trigger

* debug

* debug

* debug

* fix dir typo

* quick fix

* kubeconfig changes

* fix input

* Check

* check export

* revert kubeconfig

* Update makefile and change action input

* Fix make target in workflow, remove api gateway from api gateway prereq

* Improvements

* Refactor

* Fix typo

* fix

* fix

* finally

* fix

* fix

* typo

* Refactor

* test

* revert after test

* change target for test

* Set proper trigger

kolodziejczak committed:
fix workflow bugs (#573)

Commit:
Merge branch 'mod-dev' into default-gateway

werdes72 committed:
Add gateway disabling test

werdes72 committed:
Add gateway disabling test

werdes72 committed:
Add test for apirule blocking gateway removal

werdes72 committed:
Add test for apirule blocking gateway removal

werdes72 committed:
Merge branch 'default-gateway-integration-tests' of https://github.com/werdes72/api-gateway into default-gateway-integration-tests
